src/objectSampler.cpp

```diff
@@ -100,6 +100,10 @@ class LiveRefs {
         jvmtiEnv* jvmti = VM::jvmti();
         Profiler* profiler = Profiler::instance();
 
+        // Set CallTraceStorage counters to zero, so only live objects
+        // will be reported to collapsed and flamegraph outputs.
+        profiler->resetCounters();
```

Comment:
The idea is to reset counters only for non-JFR recording.
In JFR recording, each sample is recorded individually, so accumulated counters are not actually used.